src/Microsoft.VisualStudio.Editor.Razor/DefaultTagHelperResolver.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Razor.Extensions;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal class DefaultTagHelperResolver : TagHelperResolver
+    {
+        private readonly ErrorReporter _errorReporter;
+
+        // Hack for testability. The view component visitor will normally just no op if we're not referencing
+        // an appropriate version of MVC.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        internal bool ForceEnableViewComponentDiscovery { get; set; }
+
+        public DefaultTagHelperResolver(ErrorReporter errorReporter)
+        {
+            if (errorReporter == null)
+            {
+                throw new ArgumentNullException(nameof(errorReporter));
+            }
+
+            _errorReporter = errorReporter;
+        }
+
+        public override async Task<TagHelperResolutionResult> GetTagHelpersAsync(Project project, CancellationToken cancellationToken)
+        {
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            try
+            {
+                var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var result = GetTagHelpers(compilation);
+                return result;
+            }
+            catch (Exception exception)
+            {
+                _errorReporter.ReportError(exception, project);
+
+                throw new InvalidOperationException(
+                    Resources.FormatUnexpectedException(
+                        typeof(DefaultTagHelperResolver).FullName,
+                        nameof(GetTagHelpersAsync)),
+                    exception);
+            }
+        }
+
+        public override TagHelperResolutionResult GetTagHelpers(Compilation compilation)
+        {
+            var descriptors = new List<TagHelperDescriptor>();
+
+            var providers = new ITagHelperDescriptorProvider[]
+            {
+                new DefaultTagHelperDescriptorProvider() { DesignTime = true, },
+                new ViewComponentTagHelperDescriptorProvider() { ForceEnabled = ForceEnableViewComponentDiscovery },
```

Comment:
Does this have a `DesignTime` flag also?

Reply:
No, there's no design time specific code in view component TagHelper discovery